roles/kustomize_deploy/tasks/00_check_requirements.yml

```diff
@@ -0,0 +1,60 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Set some facts
+  ansible.builtin.set_fact:
+    _fail_message: >
+      You need to properly set the `cifmw_architecture_va_scenario` variable
+      in order to select the VA scenario to deploy. You can take a list of
+      scenario in the `examples/va` folder in the architecture repo.
+
+- name: Ensure that architecture repo exists
+  ansible.builtin.git: # noqa: latest
+    repo: "{{ cifmw_kustomize_deploy_architecture_repo_url }}"
+    dest: "{{ cifmw_kustomize_deploy_architecture_repo_dest_dir }}"
+    update: false
+    version: "{{ cifmw_kustomize_deploy_architecture_repo_version }}"
+
+- name: Ensure that the selected scenario exists
+  block:
+    - name: Create a temp variable for va scenario folder
+      ansible.builtin.set_fact:
+        _va_scenario_dir: >-
```

Comment:
Why setting a fact, that is visible everywhere, if you can use a cheap variable in the block? Same goes for https://github.com/openstack-k8s-operators/ci-framework/pull/1045/files#diff-f03da44c47ae29260b383b42de302ee733bab6fd18492aad656ef5e3595e8109R18

Reply:
makes totally sense

(This comment made me spot a mistake on line 54)